libraries/logstash_service_base.rb

```diff
@@ -20,10 +20,33 @@ class LogstashServiceBase < Chef::Resource
     default_action :start
     allowed_actions :start, :stop, :restart
 
+    def provider(arg = nil)
```

Comment:
This method looks really suspicious. Why you're patching it this way?
What chef version do you use?